### PR TITLE
Add Flapping Alerts report

### PR DESCRIPTION
**Pull request by dmitryilyin**
Add a report API and Alert List panel extension to show the list of alerts which have triggered the most number of times recently. It help to locate and fix those nasty flapping alerts.

### CODE REVIEW

File: pkg/api/alerting.go [hunk 1]
```diff
@@ -46,6 +47,36 @@ func GetAlertStatesForDashboard(c *m.ReqContext) Response {
 	return JSON(200, query.Result)
 }
 
+//GET /api/alerts/top
+func GetTopAlerts(c *m.ReqContext) Response {
```
Comment (bergquist) [inline pkg/api/alerting.go]: `TopAlerts` is kind of vague. How about `RulesWithMostStatechanges`?

File: pkg/services/sqlstore/alert.go [hunk 2]
```diff
@@ -372,3 +373,45 @@ func GetAlertStatesForDashboard(query *m.GetAlertStatesForDashboardQuery) error
 
 	return err
 }
+
+func GetTopAlerts(query *m.GetTopAlertsQuery) error {
```
Comment (bergquist) [inline pkg/services/sqlstore/alert.go]: Great! Could you please add some tests for this?

**Review by bergquist — COMMENTED** [review]
Good work!

But I would like a more descriptive name then top. How about
* Most triggered
* Most Statechanges
* Most Active
?

**Comment by dmitryilyin** [comment]
Ok, i'll change the description, url and names

**Comment by dmitryilyin** [comment]
> Good work!
> 
> But I would like a more descriptive name then top. How about
> 
> * Most triggered
> * Most Statechanges
> * Most Active
>   ?

Hello, what did you mean by "statechanges", how is it different from "triggering"? What is "active"?

**Comment by torkelo** [comment]
Not sure this warrants a completely new api and sql query. Seems like a sorting mode in the regular alert list / search api.

**Pull request closed without merging**